src/engine/execute.js

```diff
@@ -500,65 +399,49 @@ const execute = function (sequencer, thread) {
 
         // Fields are set during opCached initialization.
 
-        // Blocks should glow when a script is starting,
-        // not after it has finished (see #1404).
-        // Only blocks in blockContainers that don't forceNoGlow
-        // should request a glow.
-        if (!blockContainer.forceNoGlow) {
-            thread.requestScriptGlowInFrame = true;
-        }
-
         // Inputs are set during previous steps in the loop.
 
         const primitiveReportedValue = blockFunction(argValues, blockUtility);
 
         // If it's a promise, wait until promise resolves.
         if (isPromise(primitiveReportedValue)) {
-            handlePromise(primitiveReportedValue, sequencer, thread, opCached, lastOperation);
-
-            // Store the already reported values. They will be thawed into the
-            // future versions of the same operations by block id. The reporting
-            // operation if it is promise waiting will set its parent value at
-            // that time.
-            thread.justReported = null;
-            currentStackFrame.reporting = ops[i].id;
-            currentStackFrame.reported = ops.slice(0, i).map(reportedCached => {
-                const inputName = reportedCached._parentKey;
-                const reportedValues = reportedCached._parentValues;
-
-                if (inputName === 'BROADCAST_INPUT') {
-                    return {
-                        opCached: reportedCached.id,
-                        inputValue: reportedValues[inputName].BROADCAST_OPTION.name
-                    };
-                }
+            // Resume thread after the promise resolves
+            const generation = thread.generation;
+            primitiveReportedValue
+                .catch(rejectionReason => {
+                    // Promise rejected: the primitive had some error.
+                    // Log it and proceed.
+                    log.warn('Primitive rejected promise: ', rejectionReason);
+                    // Return an empty string
+                    return '';
+                }).then(resolvedValue => {
```

Comment:
`catch` on PromiseLike seems wrong.

p.s.: Turbowarp also had this issue before.

Reply:
Nothing new in this pull request, it's code that exists in the existing repo: https://github.com/scratchfoundation/scratch-vm/blob/16630ee9dd11b8b939c1a5db1c7ee991cb21b058/src/engine/execute.js#L138-L144

Almost certainly out of scope for this pull request to address.

Reply:
It's not.

PromiseLike (`isPromise`) only provides you `then` with `onfulfilled` and `onrejected` parameter, if you are using `catch` method, you are assuming that the PromiseLike will only be `Promise` without assertions.

Original code didn't use `catch`.

Reply:
Oh, thank you for the correction, that's a good point.

Note [`isPromise`](https://github.com/scratchfoundation/scratch-vm/blob/16630ee9dd11b8b939c1a5db1c7ee991cb21b058/src/engine/execute.js#L31-L37) is defined in this file, but it probably shouldn't be altered, because it is the definition for how block modules are expected to provide promises. Changing `isPromise` could break compatibility with existing extensions (whether part of official scratch-vm or not) and it probably isn't worth it.

Reply:
Fixed this by replacing it with the 2-argument `then` syntax.